### PR TITLE
Fixed integer overflow at stonith_admin with -t options

### PR DESCRIPTION
**Pull request by WhiteWLf-dev**

### CODE REVIEW

File: tools/stonith_admin.c [hunk 1]
```diff
@@ -528,6 +528,12 @@ main(int argc, char **argv)
         target = options.unregister_level;
     }
 
+    if (options.timeout > INT_MAX / 1000 || options.timeout < INT_MIN / 1000) {
+        printf("stonith_admin: Integer value \"%d\" for -t out of range\n", options.timeout);
+        exit_code = CRM_EX_USAGE;
+        goto done;
+    }
```
Comment (wenningerk) [inline tools/stonith_admin.c:535]: Guess we should do the error output more inline with how it is done for other errors.
Is there a sense in a negative timeout?

**Comment by clumens** [comment]
This is a little bit larger of a problem than it first appears.  We don't do a very good job of consistently handling timeouts in our command line tools - cibadmin, crm_resource, crmadmin, and stonith_admin all have a `--timeout` parameter and all handle things slightly differently.

We have a fairly old upstream issue for dealing with this inconsistency (https://projects.clusterlabs.org/T165) and there's been at least one stab at fixing the issue before.  I think it's about time we deal with that project.

File: tools/stonith_admin.c [hunk 2]
```diff
@@ -528,6 +528,12 @@ main(int argc, char **argv)
         target = options.unregister_level;
     }
 
+    if (options.timeout > INT_MAX / 1000 || options.timeout < INT_MIN / 1000) {
```
Comment (kgaillot) [inline tools/stonith_admin.c:531]: This is actually passed to functions as an `unsigned int` argument, so we should check

```
((options.timeout > (UINT_MAX / 1000)) || (options.timeout < 0))
```

File: tools/stonith_admin.c [hunk 3]
```diff
@@ -528,6 +528,12 @@ main(int argc, char **argv)
...
+        printf("stonith_admin: Integer value \"%d\" for -t out of range\n", options.timeout);
```
Comment (kgaillot) [inline tools/stonith_admin.c:532]: Simply replace `printf` with `out->err` (that formats the message as text or XML depending on the value of `--output-as`)

We don't need the `stonith_admin: ` prefix

**Review by kgaillot — CHANGES_REQUESTED** [review]
Thanks for catching this bug and many thanks for submitting a fix :)

We do need to make command-line timeout options more consistent, but this will be a good step in the meantime

**Comment by kgaillot** [comment]
ok to test

(that's for the automated CI testing)

**Review by kgaillot — COMMENTED** [review]
Looks great, thanks! I forgot one other change: could you update the commit message to "Fix: tools: validate stonith_admin --timeout value"? That way it will automatically land in the next release's ChangeLog.

**Review by kgaillot — APPROVED** [review]

**Comment by kgaillot** [comment]
FYI, I backported this to the 3.0 branch for the forthcoming 3.0.0-rc3 release

**Pull request merged**